tests/test_optimizer.py

```diff
@@ -118,87 +201,31 @@ def test_eliminate_subqueries(self):
-                "l_tax": "float64",
-                "l_returnflag": "string",
-                "l_linestatus": "string",
-                "l_shipdate": "date32",
-                "l_commitdate": "date32",
-                "l_receiptdate": "date32",
-                "l_shipinstruct": "string",
-                "l_shipmode": "string",
-                "l_comment": "string",
-            },
-            "orders": {
-                "o_orderkey": "uint64",
-                "o_custkey": "uint64",
-                "o_orderstatus": "string",
-                "o_totalprice": "float64",
-                "o_orderdate": "date32",
-                "o_orderpriority": "string",
-                "o_clerk": "string",
-                "o_shippriority": "int32",
-                "o_comment": "string",
-            },
-            "customer": {
-                "c_custkey": "uint64",
-                "c_name": "string",
-                "c_address": "string",
-                "c_nationkey": "uint64",
-                "c_phone": "string",
-                "c_acctbal": "float64",
-                "c_mktsegment": "string",
-                "c_comment": "string",
-            },
-            "part": {
-                "p_partkey": "uint64",
-                "p_name": "string",
-                "p_mfgr": "string",
-                "p_brand": "string",
-                "p_type": "string",
-                "p_size": "int32",
-                "p_container": "string",
-                "p_retailprice": "float64",
-                "p_comment": "string",
-            },
-            "supplier": {
-                "s_suppkey": "uint64",
-                "s_name": "string",
-                "s_address": "string",
-                "s_nationkey": "uint64",
-                "s_phone": "string",
-                "s_acctbal": "float64",
-                "s_comment": "string",
-            },
-            "partsupp": {
-                "ps_partkey": "uint64",
-                "ps_suppkey": "uint64",
-                "ps_availqty": "int32",
-                "ps_supplycost": "float64",
-                "ps_comment": "string",
-            },
-            "nation": {
-                "n_nationkey": "uint64",
-                "n_name": "string",
-                "n_regionkey": "uint64",
-                "n_comment": "string",
-            },
-            "region": {
-                "r_regionkey": "uint64",
-                "r_name": "string",
-                "r_comment": "string",
-            },
-        }
+        self.check_file(
+            "tpc-h/tpc-h", optimizer.optimize, schema=self.tpch_schema, pretty=True
+        )
+
+    def test_tpch_duckdb(self):
+        conn = duckdb.connect()
 
-        self.check_file("tpc-h", optimizer.optimize, schema=schema, pretty=True)
+        def replace_table(sql):
+            expression = sqlglot.parse_one(sql)
+            for table in self.tpch_schema:
+                expression = expression.with_(
```

Comment:
Why not just load all the tables into duckdb up front?
Something feels strange about mutating the queries we're about to test.

Reply:
good idea